docs/style-guide.rst

```diff
@@ -51,90 +50,558 @@ into the making of Hy.
+
+    (foo (, arg1
+            arg2
+
+    (foo (, arg1
+      arg2
+
+    ;; Judging by indentation, this is where the brackets should go.
+
+    (foo (, arg1
+            arg2))
+
+    (foo (, arg1)  ; not what we started with, is it?
+      arg2)
+
+    ;; No, it's not at all obvious it should have gone the other way.
+
+    (fn [arg
+      arg
+
+    (fn [arg]
+      arg)
+
+    ;; Beware of brackets with reader syntax. You still have to indent past them.
+
+    ;; NO!
+    `#{(foo)
+     ~@[(bar)
+      1 2]}
+
+    ;; Good.
+    `#{(foo)
+       ~@[(bar)
+          1
+          2]}
+
+3. New lines must *never* be indented past the previous element's opening bracket.
+
+   .. code-block:: clj
+
+    ;; BAD.
+    ((get-fn q)
+      x
+      y)
+
+    ;; The above with trailing brackets removed. See the problem?
+    ((get-fn q
+      x
+      y
+
+    ;; By indentation, this is where the brackets should go.
+    ((get-fn q
+      x
+      y))
+
+    ;; acceptable
+    ((get-fn q) x  ; the ")" on this line isn't trailing.
+                y)
+
+    ;; preferred, since the ) should end the line.
+    ((get-fn q)
+     x
+     y)
+
+Furthermore
+-----------
+
 + Avoid trailing spaces. They suck!
 
-+ Indentation shall be 2 spaces (no hard tabs), except when matching
-  the indentation of the previous line.
++ Limit lines to 100 characters.
+
++ Line up arguments to function calls when splitting over multiple lines.
 
   .. code-block:: clj
 
-     ;; Good (and preferred)
+    (foofunction arg1
+                 (barfunction bararg1
+                              bararg2
+                              bararg3)  ; aligned with bararg1
+                 arg3)
+
+    (foofunction arg1
+                 (barfunction bararg1
+                   bararg2)  ; Wrong. Looks like a macro body.
+                 arg3)
+
+    (foofunction arg1
+                 (barfunction bararg1 bararg2 bararg3)  ; acceptable.
+                 arg3)
+
+    ;; indenting one space past the parent bracket is acceptable for long lines
+    (foofunction
+      arg1  ; acceptable, but better to keep it on the same line as foofunction
+      (barfunction
+        bararg1  ; indent again
+        bararg2
+        bararg3)
+      arg3)  ; aligned with arg1
+
++ If you need to separate a bracket trail use a ``#_ /`` comment to hold it open.
+  (This avoids violating law #1.)
+
+  .. code-block:: clj
+
+    ;; There are basically two reasons to do this--long lists under version control,
+    ;; and when commenting out the final element during testing.
+    ;; (Common Lisp might use #+(or) for this.)
+
+    ;; preferred
+    [(foo)
+     (bar)
+     (baz)]
+
+    ;; Acceptable if the list is long. (Three isn't that long though.)
+    ;; This is better for version control line diffs.
+    [
+     (foo)
+     (bar)
+     (baz)
+     #_ /]
+
+    ;; Unacceptable and an syntax error. Lost a bracket.
+    [(foo)
+     ;; (bar)
+     ;; (baz)]
+
+    ;; Unacceptable. Broke law #1.
```

Comment:
Maybe be explicit that these few examples are for commenting out list items? Took me a few moments to realize it.